gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/offchainConfig/commit.ts

```diff
@@ -19,12 +23,77 @@ export default class CommitOffchainConfig extends SolanaCommand {
     this.require(!!this.flags.state, 'Please provide flags with "state"')
   }
 
+  makeInput = (userInput: any): Input => {
+    if (userInput) return userInput as Input
+    const rdd = getRDD(this.flags.rdd)
+    return WriteOffchainConfig.makeInputFromRDD(rdd, this.flags.state)
+  }
+
+  serializeOffchainConfig = async (
+    input: Input,
+    secret?: string,
+  ): Promise<{ offchainConfig: Buffer; randomSecret: string }> => {
+    const { configPublicKeys, ...validInput } = input
+    const proto = new Proto.Protobuf({ descriptor: OCR2Descriptor })
+    const reportingPluginConfigProto = proto.encode(
+      'offchainreporting2_config.ReportingPluginConfig',
+      validInput.reportingPluginConfig,
+    )
+    const { sharedSecretEncryptions, randomSecret } = await this.generateSecretEncryptions(configPublicKeys, secret)
+    const offchainConfig = {
+      ...validInput,
+      offchainPublicKeys: validInput.offchainPublicKeys.map((key) => Buffer.from(key, 'hex')),
+      reportingPluginConfig: reportingPluginConfigProto,
+      sharedSecretEncryptions,
+    }
+    return {
+      offchainConfig: Buffer.from(proto.encode('offchainreporting2_config.OffchainConfigProto', offchainConfig)),
+      randomSecret,
+    }
+  }
+
+  // constructs a SharedSecretEncryptions from
+  // a set of SharedSecretEncryptionPublicKeys, the sharedSecret, and a cryptographic randomness source
+  generateSecretEncryptions = async (
+    operatorsPublicKeys: string[],
+    secret?: string,
+  ): Promise<{ sharedSecretEncryptions: sharedSecretEncryptions.SharedSecretEncryptions; randomSecret: string }> => {
+    const gauntletSecret = process.env.SECRET
+    const path = join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/bip-0039', 'english.txt')
+    const randomSecret = secret || (await sharedSecretEncryptions.generateSecretWords(path))
+    return {
+      sharedSecretEncryptions: sharedSecretEncryptions.makeSharedSecretEncryptions(
+        gauntletSecret!,
+        operatorsPublicKeys,
+        randomSecret,
+      ),
+      randomSecret,
+    }
+  }
+
   makeRawTransaction = async (signer: PublicKey) => {
     const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
     const address = ocr2.programId.toString()
     const program = this.loadProgram(ocr2.idl, address)
 
+    const input = this.makeInput(this.flags.input)
     const state = new PublicKey(this.flags.state)
+    const userSecret = this.flags.secret
+    this.require(userSecret, 'Please provide the secret flag with the secret generated at write time')
```

Comment:
Secret is requirement now in commit command, so it actually verifies that the pending config is exactly the same as the one generated